### PR TITLE
feat: create with vue-i18n

### PR DESCRIPTION
**Pull request by Procrustes5**
### Description

issue : https://github.com/vuejs/create-vue/issues/545

Added vue-i18n to the option. I made it with the directory structure suggested in the above issue.

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+    <mapping directory="$PROJECT_DIR$/playground" vcs="Git" />
+  </component>
+</project>
```
Comment (cexbrayat) [inline .idea/vcs.xml:7]: can you remove all the `.idea` files that you added to this commit?

File: template/config/vue-i18n/src/i18n/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const messages = {
+  en: {
+    // English Transtlations
+  }
+}
+
+export default createI18n({
+  legacy: false,
+  globalInjection: true,
```
Comment (cexbrayat) [inline template/config/vue-i18n/src/i18n/index.js:11]: I thnink this is the default value, so the line can be removed

File: template/config/vue-i18n/src/i18n/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  locale: 'en',
+  fallbackLocale: 'en',
+  messages
+})
```
Comment (cexbrayat) [inline template/config/vue-i18n/src/i18n/index.js:15]: Also, when used in TS, there are additional informations that should be added to have a proper typechecking:
https://vue-i18n.intlify.dev/guide/advanced/typescript.html

**Review by cexbrayat — COMMENTED** [review]
This is a good start.

There should be at least one snapshot test to check that everything works.
Ideally, the should introduced a modified component template to use a translation that could be checked in a test.

**Comment by Procrustes5** [comment]
@cexbrayat 
Thank you for review!

> Ideally, the should introduced a modified component template to use a translation that could be checked in a test.

I've got it !
Is it okay to do minimal component testing first, and then plan to test all possible cases later? Of course, I want to keep responding continuously.

**Comment by Procrustes5** [comment]
@cexbrayat 
https://github.com/vuejs/create-vue-templates/pull/4
I wrote snapshot test case. I'd appreciate it if you could check this for me.

**Comment by cexbrayat** [comment]
@Procrustes5 Sorry, I should have explained: to add a snapshot test, you don't have to open a PR on create-vue-templates, you have to update the `snaphot.mjs` script in this PR. `create-vue-templates` is then automatically updated when we release a new version of create-vue.

**Comment by Procrustes5** [comment]
@cexbrayat 
Thank you!
I've changed snapshot. plz check this!

File: scripts/snapshot.mjs [hunk 4]
```diff
@@ -15,7 +15,8 @@ const featureFlags = [
   'vitest',
   'cypress',
   'playwright',
-  'nightwatch'
+  'nightwatch',
+  'vue-i18n'
```
Comment (cexbrayat) [inline scripts/snapshot.mjs:19]: I think it should be removed from here because it's going to generate too many snapshots

File: template/tsconfig/vue-i18n/i18n.d.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+ * you need to import the some interfaces
+ */
+import { DefineLocaleMessage } from 'vue-i18n'
+
+declare module 'vue-i18n' {
+  // define the locale messages schema
+  export interface DefineLocaleMessage {
+    hello: string
+  }
+}
```
Comment (cexbrayat) [inline template/tsconfig/vue-i18n/i18n.d.ts:11]: I don't think this file should be necessary? With TypeScript, you need to do something like:

```ts
import { createI18n } from 'vue-i18n'
import enUS from './locales/en-US.json'

// Type-define 'en-US' as the master schema for the resource
type MessageSchema = typeof enUS

const i18n = createI18n<[MessageSchema], 'en-US'>({
  locale: 'en-US',
  messages: {
    'en-US': enUS
  }
})
```

File: template/tsconfig/vue-i18n/tsconfig.i18n.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../base/tsconfig.app.json",
+  "compilerOptions": {
+    "composite": true,
+    "tsBuildInfoFile": "./node_modules/.tmp/tsconfig.vue-i18n.tsbuildinfo",
+    "types": ["i18n"]
+  },
+  "include": ["src/template/config/vue-i18n/**/*", "src/template/config/vue-i18n/**/*.json"]
+}
```
Comment (cexbrayat) [inline template/tsconfig/vue-i18n/tsconfig.i18n.json:9]: And i think this file can be removed as well?

**Comment by Procrustes5** [comment]
@cexbrayat 
Thank you for review!
You mean `template/tsconfig/vue-i18n` is unnecessary, is it correct?
I have changed, please check it one more time!

**Comment by cexbrayat** [comment]
@Procrustes5 
As I was pointing out, when using `vue-i18n`, the generated `src/i18n/index.ts` file should look like this I think:
```ts
import { createI18n } from 'vue-i18n'
import enUS from './locales/en-US.json'

type MessageSchema = typeof enUS

const i18n = createI18n<[MessageSchema], 'en-US'>({
  locale: 'en-US',
  messages: {
    'en-US': enUS
  }
})
```

Also, as a second step, the generated HelloWorld component could show the usage of `t()` (for example https://github.com/vuejs/create-vue/blob/main/template/code/default/src/components/HelloWorld.vue could have `You’ve successfully created a project with` translated). And the generated unit test could check this to ensure the addition of `vue-i18n` works well.

**Comment by cexbrayat** [comment]
Oh and I forgot something: the Pr should also update `ci.yml` to actually test the `vue-i18n` flag.
We don't want to test with all combinations, but maybe you could do the same that was done for the `devtools` flag.

**Comment by Procrustes5** [comment]
@cexbrayat 
Thank you so much for the easy explanation. I've modified it, so please check it out!

File: template/code/typescript-i18n/src/App.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    margin: 0 2rem 0 0;
+  }
+
+  header .wrapper {
+    display: flex;
+    place-items: flex-start;
+    flex-wrap: wrap;
+  }
+}
+</style>
```
Comment (cexbrayat) [inline template/code/typescript-i18n/src/App.vue:47]: Is this file necessary? If it doesn't change from the `typescript-default` versions, then I think it should not be duplicated if possible (same for the other components).

File: template/code/typescript-i18n/src/components/HelloWorld.vue [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+}>()
+
+const { t } = useI18n()
+</script>
+
+<template>
+  <div class="greetings">
+    <h1 class="green">{{ msg }}</h1>
+    <h3>
+      {{ t('greetings.message') }}
```
Comment (cexbrayat) [inline template/code/typescript-i18n/src/components/HelloWorld.vue:15]: 👍

**Review by cexbrayat — COMMENTED** [review]
Nice work. I'm just wondering if the components which are the same in the `typescript-default` template could not be removed to avoid duplication. Other than that, looks good to me.

File: template/code/typescript-i18n/src/App.vue (hunk 7 above)
Comment (Procrustes5) [reply template/code/typescript-i18n/src/App.vue]: @Procrustes5 
Oh, I'd not understood this architecture yet. Thank you
modified now!

**Comment by Procrustes5** [comment]
@cexbrayat 
It has been left here for a long time in a state where I can't proceed with anything. Is there anything I can do?
